src/lib/number.jl

```diff
@@ -71,28 +23,35 @@ end
 @adjoint conj(x::Number) = conj(x), r̄ -> (conj(r̄),)
 @adjoint imag(x::Number) = imag(x), ī -> (real(ī)*im,)
 
-DiffRules._abs_deriv(x::Complex) = x/abs(x)
-
- # adjoint for Fastmath operations
-for (f, fastf) in fast_op
-  if DiffRules.hasdiffrule(:Base, f, 1)
-    dx = DiffRules.diffrule(:Base, f, :x)
-    Δ = :Δ
-    if f in [:abs, :abs2]
-      Δ = :(real($Δ))
-    else
-      dx = :(conj($dx))
-    end
-    @eval begin
-      @adjoint Base.FastMath.$fastf(x::Number) =
-        Base.FastMath.$fastf(x), Δ -> ($Δ * make_fastmath($dx),)
-    end
-  elseif DiffRules.hasdiffrule(:Base, f, 2)
-    dx, dy = DiffRules.diffrule(:Base, f, :x, :y)
-    @eval begin
-      @adjoint Base.FastMath.$fastf(x::Number, y::Number) =
-        Base.FastMath.$fastf(x, y),
-        Δ -> (Δ * make_fastmath(conj($dx)), Δ * make_fastmath(conj($dy)))
-    end
-  end
-end
+# we intentionally define these here rather than falling back on ChainRules.jl
+# because ChainRules doesn't really handle nonanalytic complex functions
```

Comment:
This should probably be phrased differently. We don't currently handle non-analytic complex functions, but it's not a massive change to make it work properly with them. You could refer to [this issue that I just opened about this](https://github.com/JuliaDiff/ChainRulesCore.jl/issues/159)